components/button/floating-buttons.js

```diff
@@ -0,0 +1,236 @@
+import 'lit-media-query/lit-media-query.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { ButtonMixin } from './button-mixin.js';
+import { buttonStyles } from './button-styles.js';
+import { labelStyles } from '../typography/styles.js';
+
+class FloatingButtons extends ButtonMixin(LitElement) {
+
+	constructor() {
+		super();
+		this._container = null;
+		this._isRTL = false;
+		this._spacer = null;
+		this.minHeight = '500px';
+	}
+
+	static get properties() {
+		return {
+			/**
+			 * Whether always enable floating buttons. Note: by default, buttons may not float depending on
+			 * space available since floating buttons may result in poor user experience when view-port is small
+			 * (ex. phones).
+			 */
+			alwaysFloat: {
+				reflect: true,
+				type: Boolean,
+				attribute: 'always-float'
+			},
+
+			/**
+			 * Minimum height of view-port in order for buttons to float.
+			 */
+			minHeight: {
+				reflect: true,
+				type: String,
+				attribute: 'minheight'
+			},
+
+			_viewportIsAtLeastMinHeight: {
+				reflect: true,
+				type: Boolean
+			}
+		};
+	}
+
+	firstUpdated() {
+		this._reposition = this._reposition.bind(this);
+		this.updateComplete.then(() => {
+
+			this._container = this.shadowRoot.querySelector('.d2l-floating-buttons-container');
+			this._spacer = this.shadowRoot.querySelector('.d2l-floating-buttons-spacer');
+
+			window.addEventListener('resize', this._reposition);
+			window.addEventListener('scroll', this._reposition);
+
+			this._isRTL = (getComputedStyle(this._container).direction === 'rtl');
+			this.updateComplete.then(() => {
+				this._reposition();
+				let prevDocumentHeight = document.body.offsetHeight;
+				setInterval(() => {
+					const documentHeight = document.body.offsetHeight;
+					if (prevDocumentHeight !== documentHeight) {
+						this._reposition();
+					}
+					prevDocumentHeight = documentHeight;
+				}, 100);
+			});
+		});
+	}
+
+	disconnectedCallback() {
+		window.removeEventListener('resize', this._reposition);
```

Comment:
you should also call `super.disconnectedCallback` since `ButtonMixin` also uses this